### PR TITLE
OVH DNS Authenticator

### PR DESCRIPTION
**Pull request by baarde**
Implement an Authenticator which can fulfill a dns-01 challenge using the OVH DNS API. Applicable only for domains using OVH DNS.

Testing Done:
 * `tox -e py27`
 * `tox -e lint`
 * Manual testing:
    * Used `certbot certonly --dns-ovh -d`, specifying a credentials file as a command line argument. Verified that a certificate was successfully obtained without user interaction.
    * Used `certbot certonly --dns-ovh -d`, without specifying a credentials file as a command line argument. Verified that the user was prompted and that a certificate was successfully obtained.
    * Used `certbot certonly -d`. Verified that the user was prompted for a credentials file after selecting dnsimple interactively and that a certificate was successfully obtained.
    * Used `certbot renew --force-renewal`. Verified that certificates
      were renewed without user interaction.
 * Negative testing:
    * Path to non-existent credentials file.
    * Credentials file with unsafe permissions (644).
    * Path to credentials file with an invalid application key.
    * Path to credentials file with an invalid application secret.
    * Path to credentials file with an invalid consumer key.
    * Path to credentials file with missing properties.
    * Domain name not registered to OVH account.

### CODE REVIEW

**Comment by pde** [comment]
Awesome, thanks for this. @zjs would you have any spare cycles for a review?

File: docs/cli-help.txt [hunk 1]
```diff
@@ -431,6 +431,8 @@ plugins:
                         using LuaDNS for DNS). (default: False)
   --dns-nsone           Obtain certificates using a DNS TXT record (if you are
                         using NS1 for DNS). (default: False)
+  --dns-ovh             Obtain certificates using a DNS TXT record (if you are
+                        using NS1 for OVH). (default: False)
```
Comment (zjs) [inline docs/cli-help.txt]: nit: small copy/paste/replace mistake (should be "OVH for DNS")

**Comment by baarde** [comment]
@zjs I can send you credentials for a test domain I own if you'd like.

**Comment by quinot** [comment]
Note that should #5530 be merged, dns_ovh_test.py will need an adjustment (see changes to the other test modules in that PR -- essentially you'll need to call self.configure() instead of manually building the mock config object).

File: certbot-dns-ovh/certbot_dns_ovh/__init__.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+* ``GET /domain/zone/*``
+* ``PUT /domain/zone/*``
+* ``POST /domain/zone/*``
+* ``DELETE /domain/zone/*``
+
+These credentials can be obtained there:
+
+* `OVH Europe <https://eu.api.ovh.com/createToken/>` (endpoint: ``ovh-eu``)
+* `OVH North America <https://ca.api.ovh.com/createToken/>` (endpoint:
```
[the remark this answers is not in the document]
Comment (baarde) [reply certbot-dns-ovh/certbot_dns_ovh/__init__.py]: Thanks for the feedback. 🙂

**Pull request merged**